### ISSUE TITLE
[Feature Request] date format in path_prefix

### ISSUE THREAD

**Issue by shigemk2** [embulk/embulk#1018]
I want to output files by embulk everyday and add datetime in output files.
For example, I want the output file name to be sample_20180530.csv when I execute embulk on March 30, 2018, but there's no function in current embulk.

Here is example yaml:

```yml
in:
  ....
out:
  type: file
  path_prefix: '/tmp/out/sample_%Y%m%d' # here
  sequence_format: ''
  formatter: {type: csv, header_line: true, charset: UTF-8, newline: CRLF}
```

The other way is using [liquid file](https://gist.github.com/hiroyuki-sato/370ca003b410fb8a8870), but I don't want use liquid file.

**Comment by sakama** [comment]
Hi.

Currently, path is created by each output plugin using path_prefix.
https://github.com/embulk/embulk/blob/v0.9.7/embulk-standards/src/main/java/org/embulk/standards/LocalFileOutputPlugin.java#L83

It might not so difficult to change each plugin. But to implement common API in Embulk core will need 
 much time.
Of course, we need to consider about timezone.

There is an another problem.
I don't think you'll be able to get file sample_20180530.csv even if we support date_format in path_prefix.
File name will become like sample_20180530_00.00.csv because file name will be generated using 3 options - `path_prefix` , `taskIndex` and `fileIndex`.
Reason is that Embulk can output multiple files at 1 execution.

**Comment by sakama** [comment]
I think there're many potential requests like some Embulk users do with liquid.

Personally, I think to support something template engine is reasonable for such purposes.
I'm not sure if liquid is the best solution or not.

**Comment by shigemk2** [comment]
https://github.com/embulk/embulk/issues/1018#issuecomment-393404268
Sorry. My explanation is not enough. I want sequence_format to be empty because I want to output only one file.
This future request would be difficult because we have to implement common API and write a lot of test codes for time zone.

**Comment by hiroyuki-sato** [comment]
Hello, @shigemk2 

> Sorry. My explanation is not enough. I want sequence_format to be empty because I want to output only one file.

#501 might help as a workaround.

**Comment by dmikurube** [comment]
@shigemk2 Thanks for your question, and sorry to be really late.

Removing the `sequence_format` is not possible per Embulk's concurrent executing nature. As @hiroyuki-sato has mentioned, making the execution non-concurrent may help a bit, though.

Let me close this issue.

**Comment by motasun** [comment]
@dmikurube @sakama 
hello, I want to download data from database to a single excel file by embulk, and according to the point above, I set my liquid to:

> exec:
>   min_output_tasks: 1
> out:
>   file_ext: "xls"
>   sequence_format: ''

but I ran into a problem that there are 1001 rows of data in the database but only 519 rows to be downloaded to the excel file.
Could you please tell me how to solve this problem?
(Sorry... I am the first time to use Embulk...)

**Comment by dmikurube** [comment]
@motasun There can be many possible reasons only from the explanation. But at least, it does not seem related to this issue #1018.

Some advice :

* Try the CSV output format, not Excel, at first. The CSV formatter is a standard plugin embedded in Embulk, and used by many people. But the Excel formatter is a third-party plugin, used by less people. If it happens even with CSV, the problem is expected to be with the input side. If it does not happen with CSV, the problem is expected to be with the output (Excel) side. Then, I'd suggest to file an Issue on the repository of the expected plugin, not this `embulk` repository.
* When you file an Issue (to the repo of a plugin), share your entire configuration, and logs. It's a general manner. Only from your comment, anyone cannot understand what is the input (you said its "database", but don't understand MySQL? PostgreSQL? DynamoDB? MongoDB?), and what is the detailed output config.

**Comment by motasun** [comment]
@dmikurube 
Thank you very much for your advice~!

> Try the CSV output format, not Excel, at first. 
I have tried CSV output fomat, and the result was right with CSV format. The demand is to export a EXCEL file. but I have tried many ways but it didn't work.

> When you file an Issue (to the repo of a plugin), share your entire configuration, and logs.
I am very sorry ...
Hear is my config, Input is PostgreSQL
![config](https://user-images.githubusercontent.com/69889383/90616688-529c9800-e240-11ea-9d44-85fd77b94b5f.png)
